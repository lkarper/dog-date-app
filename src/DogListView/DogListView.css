.DogListView__profile-pic {
    height: 300px;
    width: auto;
}

.DogListView__li {
    list-style: none;
}