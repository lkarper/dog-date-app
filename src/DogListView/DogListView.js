import React from 'react';
import { Link } from 'react-router-dom';
import './DogListView.css';

const DogListView = (props) => {

    const { profile_img_url, name, id } = props.dog;

    const image = profile_img_url 
        ?
            <img
                className='DogListView__profile-pic' 
                src={profile_img_url} 
                alt={`Avatar of the dog named ${name}.`} 
            />
        :
            <img
                className='DogListView__profile-pic' 
                src='/images/photo_not_available.png' 
                alt={`Avatar of the dog named ${name} not available.`} 
            />
    ;

    return (
        <li className='DogListView__li'>
            <div className='DogListView__dog-container'>
                {image}
                <Link
                    className='DogListView__profile-link'
                    to={`/dog-profile/${id}`}
                >
                    View {name}'s profile
                </Link>
            </div>
        </li>
    );
}

export default DogListView;