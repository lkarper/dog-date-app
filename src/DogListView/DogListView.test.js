import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import DogListView from './DogListView';

describe('DogListView component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<DogListView />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<DogListView />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
