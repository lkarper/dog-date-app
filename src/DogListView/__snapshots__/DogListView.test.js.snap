// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogListView component renders the UI as expected 1`] = `
<DogListView
  dog={
    Object {
      "id": "",
      "name": "",
      "profile_img_url": "",
    }
  }
>
  <li
    className="DogListView__li"
  >
    <p
      className="DogListView__p error"
    >
      Error: Check your connection and the URL and try again.
    </p>
  </li>
</DogListView>
`;
